src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -706,39 +691,41 @@ export default class GroupComparisonStore {
         }
     });
 
-    public readonly patientToAnalysisGroups = remoteData({
-        await: () => [
-            this._originalGroupsOverlapRemoved,
-            this.sampleSet
+    public readonly patientsVennPartition = remoteData({
+        await:()=>[
+            this._originalGroups,
+            this.patientToSamplesSet,
+            this.patientKeys
         ],
-        invoke: () => {
-            let sampleSet = this.sampleSet.result!;
-            let patientToAnalysisGroups = _.reduce(this._originalGroupsOverlapRemoved.result, (acc, next) => {
-                next.studies.forEach(study=>{
-                    const studyId = study.id;
-                    study.samples.forEach(sampleId => {
-                        let sample = sampleSet.get({studyId, sampleId});
-                        if (sample) {
-                            let groups = acc[sample.uniquePatientKey] || [];
-                            groups.push(next.uid);
-                            acc[sample.uniquePatientKey] = groups;
-                        }
-                    });
-                });
-                return acc;
-            }, {} as { [patientKey: string]: string[] });
-            return Promise.resolve(patientToAnalysisGroups);
+        invoke:()=>{
+            const partitionMap = new ComplexKeyGroupsMap<string>();
+            const patientToSamplesSet = this.patientToSamplesSet.result!;
+            const groupToPatientKeys = this._originalGroups.result!.reduce((map, group)=>{
+                map[group.uid] = _.keyBy(getPatientIdentifiers([group]).map(id=>{
+                    return patientToSamplesSet.get({ studyId: id.studyId, patientId: id.patientId })![0].uniquePatientKey;
```

Comment:
may be use `patientToSamplesSet.get(id)` since id has both `studyId` and `patientId`.

Reply:
Good point